mdformat_toc/plugin.py

```diff
@@ -166,7 +166,9 @@ def _load_headings(root: RenderTreeNode, context: RenderContext) -> HeadingTree:
         heading_text = ""
         for child in inline_token.children:
             if child.type == "text":
-                heading_text += child.content
+                heading_text += RenderTreeNode([child]).render(
+                    context.with_default_renderer_for(*RENDERERS)
+                )
```

Comment:
```suggestion
                heading_text += re.sub(r"[ \t]+", " ", child.content)
```

Would this work here, for now.

I suspect there may be a reason we don't render the way you propose, but I can't remember why right now. My suspicion is based on the fact that below we render `heading_md` precisely as proposed by you, so why not here too.

This could perhaps be improved in a future PR, in the direction you propose. That PR should probably also unify `code_inline` and `text` processing here.

Reply:
hey, sorry for the delay. I'm not  a big fan of hard-coding such rules. As an example, if someone decides to make a parser extension plugin to auto capitalize titles, this would break again, it seems very brittle to me. That being said, i agree this needs to be looked at in a bigger picture. I'll apply the proposal for now so you can decide if you already want to merge this in and tackle it later.